### PR TITLE
Adds the Floor Goblin (admin only gimmick)

### PR DESCRIPTION
**Pull request by PrimeNumb**
## About the PR 
- Adds the Floor Goblin as an admin-only gimmick antagonist. An annoying miscreant living under the floor that steals people's shoes and trips people by biting at their ankles.
- Fixes an issue with the Toggle Reveal admin-only ability (should now properly open and close the floor)
- Added functionality for disabling movement puffs (might be useful for cloakers?)

Further tweaks are planned incase this gets merged, but needs to be tested properly in a live round first.

## Why's this needed? 
Admin request. Having tested the prototype of it extensively already in live rounds, it's very fun to play and adds some extra hilarity to the round.

### CODE REVIEW

File: code/mob/living/life/canmove.dm [hunk 1]
```diff
@@ -9,9 +9,10 @@
 			owner.lying = istype(owner.buckled, /obj/stool/bed) || istype(owner.buckled, /obj/machinery/conveyor)
 			if (owner.lying)
 				owner.drop_item()
-			owner.set_density(initial(owner.density))
+			if(!(owner.mob_flags & NO_DENSITY))
+				owner.set_density(initial(owner.density))
```
Comment (PrimeNumb) [inline code/mob/living/life/canmove.dm:13]: Not great, but I'd rather avoid refactoring all the life processes and things depending on it.

File: code/modules/antagonists/floor_goblin/floor_goblin.dm [hunk 2]
```diff
@@ -0,0 +1,245 @@
+/mob/proc/make_floor_goblin()
+	if (!(ishuman(src) && src.bioHolder && src.mind))
+		return
+	var/mob/living/carbon/human/H = src
+
+	message_admins("[key_name(usr)] made [key_name(H)] a macho man.")
+	logTheThing("admin", usr, H, "made [constructTarget(H,"admin")] a macho man.")
```
Comment (Tarmunora) [inline code/modules/antagonists/floor_goblin/floor_goblin.dm:7]: errr

File: code/modules/antagonists/floor_goblin/floor_goblin.dm [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+			APPLY_MOB_PROPERTY(holder.owner, PROP_CANTMOVE, "floorbiting")
+			SPAWN_DBG(0.4 SECONDS)
+				if(holder.owner && target_human)
+					playsound(floorturf, "sound/impact_sounds/Flesh_Tear_3.ogg", 50, 1, pitch = 1.3)
+					target_human.changeStatus("weakened", 2 SECONDS)
+					target_human.force_laydown_standup()
+					holder.owner.visible_message("<span class='combat'><b>[holder.owner] bites at [target_human]'s ankles!</b></span>",\
+					"<span class='combat'><b>You bite at [target_human]'s ankles!</b></span>")
+					REMOVE_MOB_PROPERTY(holder.owner, PROP_CANTMOVE, "floorbiting")
+				SPAWN_DBG(0.4 SECONDS)
```
Comment (Tarmunora) [inline code/modules/antagonists/floor_goblin/floor_goblin.dm:118]: already in a spawn - can just sleep

File: code/mob/living/life/stuns_lying.dm [hunk 4]
```diff
@@ -65,7 +65,8 @@
 
 		if (owner.lying != lying_old)
 			owner.update_lying()
-			owner.set_density(!owner.lying)
+			if(!(owner.mob_flags & NO_DENSITY))
+				owner.set_density(!owner.lying)
```
Comment (Tarmunora) [inline code/mob/living/life/stuns_lying.dm:69]: couldn't this be handled *in* the set_density proc?

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/floor_goblin/floor_goblin.dm (hunk 2 above)
Comment (PrimeNumb) [reply code/modules/antagonists/floor_goblin/floor_goblin.dm:7]: wooops

File: code/mob/living/life/stuns_lying.dm (hunk 4 above)
Comment (PrimeNumb) [reply code/mob/living/life/stuns_lying.dm:69]: That's how I did it at first, but I felt that doing mob-specific stuff inside `set_density` (a generalized proc that ideally shouldn't really care what kind of subtype it's called on) was kind of smelly. I could just make it a regular flag too and do it inside `set_density` but it's only really needed for things that tend to jump back and forth between different density values, which is mostly just mobs iirc.

File: _std/defines/mob.dm [hunk 5]
```diff
@@ -6,8 +6,8 @@
 #define SPEECH_BLOB 4		//yes
 #define SEE_THRU_CAMERAS 8	//for ai eye
 #define IS_BONER 16			//for skeletals
-#define UNUSED_32 32
-#define UNUSED_64 64
+#define NO_MOVEMENT_PUFFS 32	// sprinting/walking does not produce any smoke puffs
+#define NO_DENSITY 64	 // for mobs that should always have 0 density
```
Comment (Tarmunora) [inline _std/defines/mob.dm:10]: could both be mob properties probably actually

File: code/mob/living/life/stuns_lying.dm (hunk 4 above)
Comment (PrimeNumb) [reply code/mob/living/life/stuns_lying.dm:69]: Yeah that's a better idea

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**